### PR TITLE
Add clear all example to ComboBox docs

### PR DESCRIPTION
**Pull request by ThusharaJ07**
Implements #3307

### CODE REVIEW

File: packages/core/stories/combo-box/combo-box.stories.tsx [hunk 1]
```diff
@@ -702,3 +702,93 @@ export const FreeText: StoryFn<ComboBoxProps> = (args) => {
...
+      {...args}
+      multiselect
+      selected={selected}
+      onChange={handleChange}
+      onSelectionChange={handleSelectionChange}
+      value={value}
+      style={{ width: "266px" }}
+    >
+      {filteredOptions.length > 1 && (
+        <Option
```
Comment (joshwooding) [inline packages/core/stories/combo-box/combo-box.stories.tsx:777]: Select All should be in it's own <OptionGroup> which will allow you to remove the styles.

File: packages/core/stories/combo-box/combo-box.stories.tsx [hunk 2]
```diff
@@ -702,3 +702,93 @@ export const FreeText: StoryFn<ComboBoxProps> = (args) => {
...
+    setValue(value);
+  };
+
+  const handleSelectionChange = (
+    event: SyntheticEvent,
+    newSelected: string[]
+  ) => {
+    let newOptionsSelected = [...newSelected];
+    //case: clear all if select all is unselected
+    if (
```
Comment (joshwooding) [inline packages/core/stories/combo-box/combo-box.stories.tsx:724]: The comment above says this check is for is all is unselected but the logic is for if all is selected and it was previously selected?

Comment (ThusharaJ07) [reply packages/core/stories/combo-box/combo-box.stories.tsx:724]: corrected the comment, it for case where all select all was previously selected, if user unselects any option, then select all should be unselected

File: packages/core/stories/combo-box/combo-box.stories.tsx (hunk 1 above)
Comment (joshwooding) [reply packages/core/stories/combo-box/combo-box.stories.tsx:774]: Sorry a word got omitted, OptionGroup should be used here

File: packages/core/stories/combo-box/combo-box.stories.tsx (hunk 2 above)
Comment (mark-tate) [reply packages/core/stories/combo-box/combo-box.stories.tsx:724]: Whenever we need comments  to explain code, it's usually a sign that the code does not explain itself and there is potential to refactor.

As these code are docs, it's worthwhile to consider how we could make the code, explain itself.

```
  const handleSelectionChange: DropdownProps["onSelectionChange"] = (
    event,
    newSelected
  ) => {
    let newOptionsSelected = [...newSelected];
    const wasAllSelected = selected.includes(allSelectedOptionValue);
    const isAllSelected = newOptionsSelected.includes(allSelectedOptionValue);

    if (wasAllSelected) {
      if (isAllSelected) {
        newOptionsSelected = newOptionsSelected.filter(
          (el) => el !== allSelectedOptionValue
        );
      } else {
        newOptionsSelected = [];
      }
    } else if (
      isAllSelected ||
      (!isAllSelected && newOptionsSelected.length === usStates.length)
    ) {
      newOptionsSelected = [...usStates, allSelectedOptionValue];
    }
    setSelected(newOptionsSelected);
    args.onSelectionChange?.(event, newOptionsSelected);
  };
```

How about this, any easier to read?

Comment (ThusharaJ07) [reply packages/core/stories/combo-box/combo-box.stories.tsx]: yes its easier! I have updated this. Thanks Mark for this.

**Comment by ThusharaJ07** [comment]
> When i use keyboard to toggle "Select All" option (https://62565e7f.saltdesignsystem-storybook.pages.dev/?path=/story/core-dropdown--select-all), focus is changed to the 2nd item. Is it the intent?

Hey @origami-z , this is an existing issue in dropdown which Shey said will have a word with the team separately. not related to this example.

**Comment by origami-z** [comment]
> Hey @origami-z , this is an existing issue in dropdown which Shey said will have a word with the team separately. not related to this example.

I can only see the issue in dropdown but not combobox, is it to do with `key` ?

File: site/docs/components/combo-box/examples.mdx [hunk 3]
```diff
@@ -176,5 +176,13 @@ When the `value` prop is an object, the `valueToString` prop must be provided to
 
 Use the `truncate` prop when you are limited on space and want to prevent the multi-select combobox from spanning multiple lines.
 
+</LivePreview>
+
+<LivePreview componentName="combo-box" exampleName="ClearAll" >
```
Comment (joshwooding) [inline site/docs/components/combo-box/examples.mdx:181]: A question to ask the group, do we want to document this as clear all? Or an example of using a end adornment

File: site/docs/components/combo-box/examples.mdx [hunk 4]
```diff
@@ -176,5 +176,13 @@ When the `value` prop is an object, the `valueToString` prop must be provided to
...
+
+## Clear All
+
+In order to achieve the clear all functionality, you can pass a close icon in endAdornment prop.
```
Comment (joshwooding) [inline site/docs/components/combo-box/examples.mdx:185]: This guidance is incorrect, a button is passed to endAdornment not an icon. Also we typically use backticks (`) when referencing props.

**Review by shey-v — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**